src/libPMacc/include/cuSTL/algorithm/kernel/run-time/Foreach.hpp

```diff
@@ -68,6 +69,18 @@ namespace RT
         /* C0 c0_shifted = c0(p_zone.offset); ...; CN cN_shifted = cN(p_zone.offset); */                              \
         BOOST_PP_REPEAT(N, SHIFT_CURSOR_ZONE, _)                                                                    \
                                                                                                                     \
+        if(this->_blockDim == math::Size_t<3>::create(0))                                                           \
```

Comment:
offline discussed: thus should be moved in a helper function, there is no need to write it in a macro.